### PR TITLE
Kube Improvements.

### PR DESCRIPTION
**Pull request by davinchia**
## What
Make sync loop more efficient with pinging the Kube server when checking pod statuses.

 Also add a counter for records read to make it clearer to users.

## How
* In the replication worker, only check if the source and destination is empty if the message iterator is empty.
* In the KubePodProcess, cache and use the last status check for 30 seconds. This means we might spend up to 30 seconds waiting for a pod to be reported as completed, but the alternative, repeatedly hitting the Kube API server is worse since it affects ALL pods in the cluster.

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in output spec
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Reference docs in the `docs/integrations/` directory.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by davinchia** [comment]
@jrhizor I tested these changes on EKS and they seem comparable to my docker (they are even slightly faster).

Since for RDS postgres -> redshift

Docker:
Screen Shot 2021-06-24 at 3 40 02 PM

Kubernetes:
Screen Shot 2021-06-24 at 3 46 58 PM

Some of the speed increase is probably data staying within AWS' network instead of crossing into my machine.

As you can see, I had 6 successful kube jobs in a row, which gives me confidence this solution fixes the issue Porter saw.

**Comment by davinchia** [comment]
Tested on a larger set and looks like it works great.

Screen Shot 2021-06-24 at 4 00 55 PM

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java [hunk 1]
```diff
@@ -475,7 +475,7 @@ public void testScheduledSync() throws Exception {
     // todo: wait for two attempts in the UI
     // if the wait isn't long enough, failures say "Connection refused" because the assert kills the
     // syncs in progress
-    sleep(Duration.ofMinutes(2).toMillis());
+    sleep(Duration.ofMinutes(4).toMillis());
```
Comment (davinchia) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java:478]: This is taking slightly longer for Kube. What was happening is we were checking this in the middle of an actual run.

Should definitely change this to properly query the api. Will do so in a separate PR.

File: airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java [hunk 2]
```diff
@@ -213,6 +214,11 @@ private static Runnable getReplicationRunnable(Source<AirbyteMessage> source,
 
             sourceMessageTracker.accept(message);
             destination.accept(message);
+            recordsRead += 1;
+
+            if (recordsRead % 100 == 0) {
```
Comment (jrhizor) [inline airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java:219]:
```suggestion
            if (recordsRead % 1000 == 0) {
```

**Review by jrhizor — APPROVED** [review]
